### ISSUE TITLE
ProDOS 2.5.0a5 date and time issues

### ISSUE THREAD

**Issue by bobbimanners** [ProDOS-8/ProDOS8-Testing#21]
ProDOS 2.5.0a5 date and time issues

BUG

2.5
## Description

This is a continuation of bug ticket #20.  Now using 2.5.0 alpha 5.

Running my test code `filedate /h1/prodos` gives the same result as on 2.5.0a4:
```
ProDOS Version 25

Creation:     d=$242d t=$0909
Modification: d=$2432 t=$0700
Legacy ProDOS 1.0 to 2.4.2 date and time:
Creation:     18-01-13 09:09
Modification: 18-01-18 07:00
ProDOS 2.5.0 date and time:
Creation:     1069-01-01 08:09
Modification: 1074-01-00 56:00
```

## Expected Behavior

The ProDOS date and time format should be the new format.

## Actual Behavior

However, we are still seeing the old format.  The output above decodes the date using the old and the new formats.  The values should for the old format are correct, the new ones are garbage.

## Possible Fix

ProDOS 2.5.0 should be returning dates in the new format in this case.

## Steps to Reproduce

1. You can run the attached `filedate` program within the Aztec C shell.
2. Run it passing in a filename on the command-line `filedate foo`
3. Compare the time and date values with those shown by `CATALOG` for example.

## Context

Alpha testing the new 2.5.0a5 to see if the date conversion code could cause other issues.

### Hardware:

Apple IIe Enhanced 128K
Disk ][ in slot 6
MicroDrive/Turbo in slot 7.

### ProDOS8 version:
2.5.0 alpha 5

### Non-Operating System release software being used: 
Aztec C65 3.2

**Comment by bobbimanners** [comment]
Will upload the attachment in a few moments . . .

**Comment by bobbimanners** [comment]
Here is the `filedate` code, as a ZIPed ShrinkIt file (!)

[filedate.shk.zip](https://github.com/ProDOS-8/ProDOS8-Testing/files/3611738/filedate.shk.zip)

**Comment by JohnMBrooks** [comment]
Confirmed in the P8 Exerciser app that issuing ProDOS MLI $C4 GetFileInfo on legacy-date files does not auto-convert to new-date format

**Comment by JohnMBrooks** [comment]
Fixed in 2.5.0a6
